pkg/actuators/machine/utils.go

```diff
@@ -406,3 +406,67 @@ func getClusterID(machine *machinev1.Machine) (string, bool) {
 	}
 	return clusterID, ok
 }
+
+// RawExtensionFromProviderSpec marshals the machine provider spec.
+func RawExtensionFromProviderSpec(spec *machinev1.AWSMachineProviderConfig) (*runtime.RawExtension, error) {
+	if spec == nil {
+		return &runtime.RawExtension{}, nil
+	}
+
+	var rawBytes []byte
+	var err error
+	if rawBytes, err = json.Marshal(spec); err != nil {
+		return nil, fmt.Errorf("error marshalling providerSpec: %v", err)
+	}
+
+	return &runtime.RawExtension{
+		Raw: rawBytes,
+	}, nil
+}
+
+// RawExtensionFromProviderStatus marshals the machine provider status
+func RawExtensionFromProviderStatus(status *machinev1.AWSMachineProviderStatus) (*runtime.RawExtension, error) {
+	if status == nil {
+		return &runtime.RawExtension{}, nil
+	}
+
+	var rawBytes []byte
+	var err error
+	if rawBytes, err = json.Marshal(status); err != nil {
+		return nil, fmt.Errorf("error marshalling providerStatus: %v", err)
+	}
+
+	return &runtime.RawExtension{
+		Raw: rawBytes,
+	}, nil
+}
+
+// ProviderSpecFromRawExtension unmarshals a raw extension into an AWSMachineProviderSpec type
+func ProviderSpecFromRawExtension(rawExtension *runtime.RawExtension) (*machinev1.AWSMachineProviderConfig, error) {
+	if rawExtension == nil {
+		return &machinev1.AWSMachineProviderConfig{}, nil
+	}
+
+	spec := new(machinev1.AWSMachineProviderConfig)
+	if err := json.Unmarshal(rawExtension.Raw, &spec); err != nil {
+		return nil, fmt.Errorf("error unmarshalling providerSpec: %v", err)
+	}
+
+	klog.V(5).Infof("Got provider Spec from raw extension: %+v", spec)
+	return spec, nil
+}
+
+// ProviderStatusFromRawExtension unmarshals a raw extension into an AWSMachineProviderStatus type
+func ProviderStatusFromRawExtension(rawExtension *runtime.RawExtension) (*machinev1.AWSMachineProviderStatus, error) {
+	if rawExtension == nil {
+		return &machinev1.AWSMachineProviderStatus{}, nil
+	}
+
+	providerStatus := new(machinev1.AWSMachineProviderStatus)
+	if err := json.Unmarshal(rawExtension.Raw, providerStatus); err != nil {
+		return nil, fmt.Errorf("error unmarshalling providerStatus: %v", err)
+	}
+
+	klog.V(5).Infof("Got provider Status from raw extension: %+v", providerStatus)
+	return providerStatus, nil
+}
```

Comment:
Do these need to be exported functions?

Reply:
I think yes, if anyone is importing our fork they might use it.

Reply:
If that is a concern, can I suggest that we put these into their own package, don't remove them from the API package but instead "deprecate" them by creating aliases? (We may want to do this for our API types too to make this transition smoother)

```
// Deprecated: Use utils.RawExtensionFromProviderSpec instead.
var RawExtensionFromProviderSpec = utils.RawExtensionFromProviderSpec
```

Reply:
We can do aliases for these functions, but API types will get outdated quickly which means that for further use people will have to switch to openshift/api anyway.